### ISSUE TITLE
integrate all built-in PyTorch architectures

### ISSUE THREAD

**Issue by sammlapp** [kitzeslab/opensoundscape#278]
Unfortunately, it is not trivial to allow the choice of any pytorch architecture as a 1-for-1 substitution - each one requires different manual steps to change the number of output nodes in the classification layer. 
We should create a wrapper or module that enables 1-for-1 swap out of the currently implemented architectures [resnet, alexnet, vgg, squeezenet, densenet, inception]. Inception has extra complications and we might choose to leave it out for now. 

See this article for detailed instructions - https://pytorch.org/tutorials/beginner/finetuning_torchvision_models_tutorial.html , especially `def initialize_model()`

**Comment by sammlapp** [comment]
things to revisit in our own code:
- update_best()
- BaseArchitecture class
(we may not need these and removing them brings us closer to simply using built-in architectures)

**Comment by sammlapp** [comment]
Notes from my commments that I'm moving here:
# Problem - each pytorch architecture needs a different mod for chaning final layer
# 1) move loss function outside of architecture
# 2) move update_best outside of architecture
# 3) don't rely on separation of feature/classifier when setting parameters
# 4) custom models have forward pass in 1 step (added .forward)
# Then we can enable other networks like this:
# https://pytorch.org/tutorials/beginner/finetuning_torchvision_models_tutorial.html

**Comment by sammlapp** [comment]
I added wrappers for a bunch of currently available architectures, but it would be better if there were a generic way to use any architecture and modify the output shape. I asked a question related to this on pytorch forums:
https://discuss.pytorch.org/t/change-output-shape-of-any-cnn-architecture/121886 

For now, there are several architectures working well so I'll merge my current branch.

**Comment by sammlapp** [comment]
resolved by #301
